tistory oauth2 연결 중에 죽는 문제.

follow-redirects version이 업데이트 되면서 아래와 같이 연결오류가 발생하는 것 같습니다.

``` js
AssertionError: callback must be a function
    at parseOptions (/var/lib/openshift/55f035ba2d527171d3000173/app-root/runtime/repo/node_modules/passport-tistory/node_modules/follow-redirects/create.js:117:12)
    at Object.H.request (/var/lib/openshift/55f035ba2d527171d3000173/app-root/runtime/repo/node_modules/passport-tistory/node_modules/follow-redirects/create.js:102:22)
    at exports.OAuth2._executeRequest (/var/lib/openshift/55f035ba2d527171d3000173/app-root/runtime/repo/node_modules/oauth/lib/oauth2.js:132:30)
    at exports.OAuth2._request (/var/lib/openshift/55f035ba2d527171d3000173/app-root/runtime/repo/node_modules/oauth/lib/oauth2.js:111:8)
    at TistoryStrategy._oauth2.getOAuthAccessToken (/var/lib/openshift/55f035ba2d527171d3000173/app-root/runtime/repo/node_modules/passport-tistory/lib/passport-tistory/strategy.js:54:10)
    at TistoryStrategy.OAuth2Strategy.authenticate (/var/lib/openshift/55f035ba2d527171d3000173/app-root/runtime/repo/node_modules/passport-tistory/node_modules/passport-oauth2/lib/strategy.js:169:18)
    at attempt (/var/lib/openshift/55f035ba2d527171d3000173/app-root/runtime/repo/node_modules/passport/lib/middleware/authenticate.js:341:16)
    at Object.authenticate [as handle] (/var/lib/openshift/55f035ba2d527171d3000173/app-root/runtime/repo/node_modules/passport/lib/middleware/authenticate.js:342:7)
    at next_layer (/var/lib/openshift/55f035ba2d527171d3000173/app-root/runtime/repo/node_modules/express/lib/router/route.js:103:13)
    at Route.dispatch (/var/lib/openshift/55f035ba2d527171d3000173/app-root/runtime/repo/node_modules/express/lib/router/route.js:107:5)
    at /var/lib/openshift/55f035ba2d527171d3000173/app-root/runtime/repo/node_modules/express/lib/router/index.js:205:24
    at Function.proto.process_params (/var/lib/openshift/55f035ba2d527171d3000173/app-root/runtime/repo/node_modules/express/lib/router/index.js:269:12)
    at next (/var/lib/openshift/55f035ba2d527171d3000173/app-root/runtime/repo/node_modules/express/lib/router/index.js:199:19)
    at next (/var/lib/openshift/55f035ba2d527171d3000173/app-root/runtime/repo/node_modules/express/lib/router/index.js:176:38)
    at Function.proto.handle (/var/lib/openshift/55f035ba2d527171d3000173/app-root/runtime/repo/node_modules/express/lib/router/index.js:151:3)
    at Layer.router (/var/lib/openshift/55f035ba2d527171d3000173/app-root/runtime/repo/node_modules/express/lib/router/index.js:24:12)
```

--- a/lib/passport-tistory/strategy.js
+++ b/lib/passport-tistory/strategy.js
@@ -3,9 +3,7 @@ var util = require('util'),
   querystring= require('querystring'),
   OAuth2Strategy = require('passport-oauth2');
 
-
-var http = require('follow-redirects').http;
-var https = require('follow-redirects').https;
+var request = require('request');
 
 /**
  * TistoryStrategy 생성자.<br/>
@@ -33,14 +31,6 @@ function TistoryStrategy(options, verify) {
   this.name = 'tistory';
   this._userProfileURL = 'https://www.tistory.com/apis/blog/info?output=json';
 
-  this._oauth2._chooseHttpLibrary= function( parsedUrl ) {
-    var http_library= https;
-    if( parsedUrl.protocol != "https:" ) {
-      http_library= http;
-    }
-    return http_library;
-  };
-
   this._oauth2.getOAuthAccessToken= function(code, params, callback) {
     var params= params || {};
     params['client_id'] = this._clientId;
@@ -51,8 +41,8 @@ function TistoryStrategy(options, verify) {
 
     var url = this._getAccessTokenUrl() + "?" + querystring.stringify(params);
 
-    this._request("GET", url, {}, "", null, function(error, data, response) {
-      if( error )  callback(error);
+    request(url, function(error, response, data) {
+      if( error ) callback(error);
       else {
         var results;
         try {

--- a/package.json
+++ b/package.json
@@ -2,9 +2,12 @@
   "name": "passport-tistory",
   "version": "0.1.0",
   "description": "Tistory authentication strategy for Passport.",
-  "author": { "name": "SungKwang Song", "email": "saltfactory@gmail.com", "url": "http://saltfactory.net/" },
-  "contributors": [
-  ],
+  "author": {
+    "name": "SungKwang Song",
+    "email": "saltfactory@gmail.com",
+    "url": "http://saltfactory.net/"
+  },
+  "contributors": [],
   "repository": {
     "type": "git",
     "url": "git://github.com/saltfactory/passport-tistory.git"
@@ -14,14 +17,24 @@
   },
   "main": "./lib/passport-tistory",
   "dependencies": {
-    "passport": "~0.2.0",
-    "passport-oauth2":"~1.1.2",
-    "follow-redirects":"~0.0.3",
-    "querystring":"~0.2.0"
+    "passport": "^0.3.0",
+    "passport-oauth2": "^1.1.2",
+    "querystring": "^0.2.0",
+    "request": "^2.64.0"
   },
-  "licenses": [ {
-    "type": "MIT",
-    "url": "http://www.opensource.org/licenses/MIT"
-  } ],
-  "keywords": ["passport", "tistory", "auth", "authn", "authentication", "identity", "saltfactory"]
+  "licenses": [
+    {
+      "type": "MIT",
+      "url": "http://www.opensource.org/licenses/MIT"
+    }
+  ],
+  "keywords": [
+    "passport",
+    "tistory",
+    "auth",
+    "authn",
+    "authentication",
+    "identity",
+    "saltfactory"
+  ]
 }

